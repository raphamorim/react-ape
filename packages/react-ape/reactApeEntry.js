/**
 * Copyright (c) 2018-present, Raphael Amorim.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import ReactApeRenderer from './renderer/reactApeRenderer';
import StyleSheetModule from './modules/StyleSheet';
import DimensionsModule from './modules/dimensions';

import ListViewComponent from './renderer/components/ListView';
export const ListView = ListViewComponent;

export const render = ReactApeRenderer.render;
// export const unmountComponentAtNode = ReactTVRenderer.unmountComponentAtNode;

export const Image = 'Image';
export const View = 'View';
export const Text = 'Text';

export const StyleSheet = StyleSheetModule;
export const dimensions = DimensionsModule;

export default ReactApeRenderer;
