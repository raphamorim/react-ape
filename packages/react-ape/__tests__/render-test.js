import React from 'react';
import { render, View, Text, StyleSheet } from '../reactApeEntry';

describe('render', () => {
  it('renders <Text/> correctly', () => {
    const styles = StyleSheet.create({
      view: {
        backgroundColor: 'green',
      },
      text: {
        color: '#F8F8F8',
        fontFamily: 'Arial'
      }
    })

    const App = (
      <View style={styles.view}>
        <Text style={styles.text}>
          My amazing text
        </Text>
      </View>
    );

    const canvas = document.createElement('canvas');
    render(App, canvas);
    const dataUrl = canvas.toDataURL();

    expect({dataUrl}).toMatchSnapshot();
  });
});
