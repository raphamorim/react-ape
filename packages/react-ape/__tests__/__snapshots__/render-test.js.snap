// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders <Text/> correctly 1`] = `
Object {
  "dataUrl": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAACWCAYAAABkW7XSAAAABmJLR0QA/wD/AP+gvaeTAAAPHElEQVR4nO3de3BU5cHH8W8um2RJgJAESaIBYhBDAEElXAOSMEUR9Z0Q7gi2QMXWqoURWwo2zIAinTpa+gJTlZoALbRBLhZFpVVJCeUSwDeJEAki5AobiCRkc2GT3fePDKtrLnKRyzP9fWbyB2efPc+zC/lyztlD8GIxLkSkZYvxutlLkG943+wFiIhcLgVLRIyhYImIMRQsETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SM4Xslg5+Nf5ZBtw+iwdnA7O2zcTgdLY5LHZ5Kz9Ce7C7azeqDq3+QhZpiyQNLcOHit7t+e0PnDQkIwe6wU99Yf332bw2hxlFDXUPdddm/yOW4oiOsgZEDSYlNYVLcJBK7J7Y4JiQghOeHPE9KbAoDIgb8IIs0SWL3RBK7tfzeXC+juo8i76k82vu1vy77T+yWSO6TuXT073hd9i9yua7oCAug0dVIZX0l43uN56MTHzV7PDk2mbqGOix+lh9kgaaZtX3WDZ9z8B2DCQkIuW77H3T7IEKtoddt/yKX64qD5XQ52frFVpLvTsbibWl2WjgxbiLbC7YzpfcU97YgvyAs3ha+rvu62f7a+7XHy8uLqvqqNucNDwwnNiwWX29fSi6UkH82H9e3ftBEB/8ONDgbqHHU0DOkJ1Edojj+9XFOVZ4CIMA3gPiIeOoa6sgrz6O2obbFOe4OvRuLj6XZHBZvC0F+Qa2uz+6wc7HxIqUXSj22f3td0cHRxHSKwWa3kWvL9Vj/JV07dKVHSA9sdht55XkEWgLx8fZp9f0JtATi7+MPQHBAMLUNtdgddvfjHf070r9Lf3y8ffjs9GdU1FW4H7P6WgnwDaC2odbjVO/Sa73YeBHAvf+O/h2xO+xUX6xu9X0QuZ6uOFgAGUcymNlvJkndk/jwxIfu7eGB4SREJZCyKcUjWM8NfI4XE15kaNpQDp0+5N4e4BtAwc8L2HpsK0+9/1SLc4VZw3jrkbd4KOYhj+3ZZdkkZyRTXlMOwIdTPuTouaME+AQwLnYcAC6Xi2V7lrHzxE42jttIl8AuAJysPMnDGx7mxPkTAIRaQ1nzyJpmcxwsO0hyRjK2GhtJ3ZPYNnFbq+/J9G3TyTiawY4pO3C5XAxfO9y9rvxz+Xh7eTMpbpJ7/L7SfSRnJFNR2xQQq6+VVWNWMTluMl5eTT/R5OOTH2N32IkOjmbAmpZPr1eNWeXeb96cPDbnb2bq1ql44cXChIXMHzLfHRyH08Fr+15jceZinC4nPUJ6sPuJ3eTYchi5diSNrkYA3hz7JhPiJvDgXx9kVv9Z7t/LnCdz2HZsG5M2T2pxLSLX21V9SphZmImtxsb4XuM9tqf0SqGitoJPTn7isX1dzjqcLqdHxADG9hhLcEAw63LWtTrXqjGrGNF1BBM3TyT01VCiVkSxOHMxAyIG8Gz8sx5jJ/SaQAf/DgxYM4C+b/RlX+k+fj3012wct5HUXan0XNWT2dtnE9UhigXDFrQ5R2pmKvdH3M+zA5vm2Fuyl6FpQz2+xm4cS1V9FXnleR7h/q6U2BQi20cyPH04MStjWL5nOYMiB/Gz+3/mHrN81HImxU3iN5/+hsjXIxn454G092vPo3c92up+AZ7/5/P88cAfARiWPoxffvRLAOYNnseihEWs+WwN0f8bTdcVXVm+ZznzB8/nhSEvAJBryyV1VyrxEfH8Iv4XAEzpPYXJvSezLGsZu4t288K/XuD1/a8DMDx9OM988Eyb6xG5nq4qWI2uRrbkb+GRux7Bz8fPvX1i3EQ252+mwdngMb6wqpBdhbuYGDcRHy8f9/apfaZy4vwJ9hTvaXEeHy8fqh3VvJT1Eu8eexe7w055TTmv7HmFM/Yz9O7cu9lzZrw7g7zyPAoqCliZvRJvL28yjmaQlpNGYVUh6/PWk382n7iwOPccdoedl7Ne9phj+Z7lnLafds9RWV/JodOH3F/Hvz7O8lHLqW2oJTkjuc1TWhcuJrwzgQNlByi5UMLizMVU1lcy+PbBQNMncDP7zWTD5xt4bd9rVNRVkGPLYcqWKd/7qVx5TTnn688DcOr8KWw1Nvx9/Jk/eD6ZhZnM2zmPsuoybDU2lu5eyrsF7zJ30Fz3UdeKAyv49NSnpA5PZUTXEbw++nX+XfRvlmUt+2b/defdv4+2Glub6xG5nq7qlBAg42gGc+6bQ1L3JD748gO6dezGwIiBLPh4QYvj0/8vnbTH0kjqnsTOr3bSuV1nRt85mmVZy1q8lgNNYZz5j5lA03WVnqE96XdbP/qH9yfQEkg7SzuP8QUVBe5TLMB9unio7JDHOFuNjcigyKua49K4DckbuDP4Tkb/dTRFVUVtvldFVUXub3poClhRVZH7U70htw/B19uXHV/u8Hhe8YViPjvzWZvXzlpyT5d7CA4I5oz9DCmxKR6PVV+spqN/R/rc1oeDZQdxupzM2j6LAzMPsGPKDs7XneeJbU+4Tw9FbiVXHaw9xXsoqy4jJTaFD778gPG9xlN8oZj/FP8Hb6/mB27bjm2jsr6SqX2msvOrne6jrfW569ucJzY0lleSXmFU9Cj3hfu9JXtb/IZq7WJwTUONx6+dLudlzfHdcZesHLOSxG6JTNs6jeyy7DbXD3Cu9lyzbY5Gh/t96hTQCYBye3mzcWfsZ644WJc+0ZvQawITek1ocwxAyYUStnyxhVn9Z/HRiY8orS5t8TkiN9tVB8vpcvJO/js83udx/Hz8GB87noyjGa0eLdU21LLp6CYmxU3C6mtlcu/JZBZmUlhV2OocnQI6sWPKDny8fZi3cx5ZRVnkn8vH6XJy/OnjzcLY2txtCQ4I5v0p72PxtjSbo+DnBc3mWJiwkBl9Z7Do00Vs+WLLFc/XkuILxQCEB4U3e6ylbd/HfrHpU8K5O+de1o278ZHxPHHPE5yqPMXkuMls/Hxjm9fkRG6Wa/qnORlHMggOCGZmv5ncG34vGUcz2hyfnpNOkF8QU/tMZUD4ANbltn6xHWB41+FEBEXw8u6XefPwmxw5ewSny0mYNYyIoAisFuu1LL9pjqjhRAZF8nKW5xyh1lAigiII8Alwj53WZxqLhi0iLSeN3+/9/TXPfUmOLQe7w97sAvsd7e+gf5f+3/v8S9cML8U1tzwXh9PB6DtHNxv79ICnWfvYWvd9W4GWQN5+9G2KLxQzJG0In5/9nDfGvuFxBPbd/YvcLNf0J3B/6X6KqopYOnIpxyqOcfj04e8dn38unyUPLKHaUc3WL7a2Ob6sugxoOgK49M3SuV1n0v8nHR8vH4+YXK1vz3HpA4Ewaxjpj6Xj6+3rjuLIbiNZPWY1WcVZLN61mOjgaI+vMGvYVa+horaCldkrGRc7jhcTXiQ6OJohtw/hb+P+5r443pYz1WcASIpOoktgF87XnWdtzlrGxIxh/pD57tf1o+gfseSBJUS2j3Tfj/W7Ub8jJjiGOe/NoaK2giffe5JQayirx3xzZHbGfsb9HoQHXvkRn8gP5apPCaHpFGzT0U3MHTSXvx/5+2U9Z13uOl4a+RJrc9d63ODYkgOlB1ift57H+zxOUvckztWeo0enHmz5Ygubjm5i9J2jsfpaW7wJ9HJll2WzLncd0/tOJ7FbYrM5Hox5EKuvlRn3zMDPx4+EqAROPnOyxdf10/d+etXrWLp7KV0Cu7Bg2AIWJizE5XKx4cgG2lnafe8nhYdOH+Ji40XSHk1jf+l+Rqwdwfx/zadTQCeWPLCEBUMXUHWxivDAcA6dPsS0rdOApttKZvWfxZ8O/Yldhbvc+3p176v8auiv+PE9PyYtJ43Dpw9T31jP24++TXZZNgnpCVf9OkWuhdeV/L+EfTr3oXO7znxy6pv7rDq360zf2/p63EXthReJ3RM5XX2aI2ePeOzjoZiH2DphK0nrk1q9neG7BkYOpF+XfjgaHWQVZ1FQUUBUhyjuCrmL7LJsquqriI+Ix4XL4yJ4iDWE/l36k2vLdX9iCHBv+L34efuxr3TfZc9xR/s72ryedOm1xkfGA02xBVpcV1vbw6xhRHeKpvRCKSUXSjjy1BG+Ov8VYzeObfM9ig6O5r7w+yivKSezMNO9ve9tfYmPaDpCzT+XT1ZRlvta333h9xEcEMy+kn0ef3n4+/gzLGoY9ot293vUvWN37o+4n7M1Z91x+6+g/5fwlnJFwfohbEzeSEynGOL/HH8jp72lWbwt5M3JY/XB1e6bNKEpErlzcvnD/j+w6NNFN3GF/8UUrFvKNZ0SXi6Lt4VlScvo2qErj/V8jNnbZ9+IaY3hcDooqCggdUQqQX5B5NhyiAiK4LmBz1HjqOGtw2/d7CWK3BJuSLAcTgcP93iYyKBIVhxYwV/y/nIjpjXK9G3TWTBsAT/p9xMigiKoqq9qugM9M5WTlSdv9vJEbgk3/JRQxCg6Jbyl6MYaETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SMoWCJiDEULBExhoIlIsZQsETEGAqWiBhDwRIRYyhYImIMBUtEjKFgiYgxFCwRMYaCJSLGULBExBgKlogYQ8ESEWMoWCJiDAVLRIyhYImIMRQsETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SMoWCJiDEULBExhoIlIsZQsETEGAqWiBhDwRIRYyhYImIMBUtEjKFgiYgxFCwRMYaCJSLGULBExBgKlogYQ8ESEWMoWCJiDAVLRIyhYImIMRQsETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SMoWCJiDEULBExhoIlIsZQsETEGAqWiBhDwRIRYyhYImIMBUtEjKFgiYgxFCwRMYaCJSLGULBExBgKlogYQ8ESEWMoWCJiDAVLRIyhYImIMRQsETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SMoWCJiDEULBExhoIlIsZQsETEGAqWiBhDwRIRYyhYImIMBUtEjKFgiYgxFCwRMYaCJSLGULBExBgKlogYQ8ESEWMoWCJiDAVLRIyhYImIMRQsETGGgiUixlCwRMQYCpaIGEPBEhFjKFgiYgwFS0SMoWCJiDEULBExhoIlIsZQsETEGAqWiBhDwRIRYyhYImIMBUtEjKFgiYgxFCwRMYaCJSLGULBExBgKlogYQ8ESEWMoWCJiDAVLRIyhYImIMRQsETHG/wNixiReChEV8AAAAABJRU5ErkJggg==",
}
`;
